components/admin_console/team_channel_settings/channel/details/channel_details.tsx

```diff
@@ -31,39 +31,39 @@ import SaveChangesPanel from '../../save_changes_panel';
 import {ChannelModes} from './channel_modes';
 import {ChannelGroups} from './channel_groups';
 import {ChannelProfile} from './channel_profile';
-import ChannelModeration from './channel_moderation';
 import ChannelMembers from './channel_members';
+import ChannelModeration from './channel_moderation';
 
-interface ChannelDetailsProps {
+export interface ChannelDetailsProps {
     channelID: string;
     channel: Channel;
     team: Partial<Team>;
     groups: Group[];
     totalGroups: number;
-    channelPermissions?: Array<ChannelPermissions>;
     allGroups: Dictionary<Group>;
+    channelPermissions?: Array<ChannelPermissions>;
     teamScheme?: Scheme;
     guestAccountsEnabled: boolean;
-    isDisabled: boolean;
+    isDisabled?: boolean;
     actions: {
-        getGroups: (channelID: string, q?: string, page?: number, perPage?: number) => Promise<Partial<Group>[]>;
-        linkGroupSyncable: (groupID: string, syncableID: string, syncableType: string, patch: Partial<SyncablePatch>) => ActionFunc|ActionResult;
-        unlinkGroupSyncable: (groupID: string, syncableID: string, syncableType: string) => ActionFunc;
+        getGroups: (channelID: string, q?: string, page?: number, perPage?: number, filterAllowReference?: boolean) => any;
+        linkGroupSyncable: (groupID: string, syncableID: string, syncableType: SyncableType, patch: SyncablePatch) => ActionFunc|ActionResult;
+        unlinkGroupSyncable: (groupID: string, syncableID: string, syncableType: SyncableType) => ActionFunc;
         membersMinusGroupMembers: (channelID: string, groupIDs: Array<string>, page?: number, perPage?: number) => ActionFunc|ActionResult;
-        setNavigationBlocked: (blocked: boolean) => any;
+        setNavigationBlocked: (blocked: boolean) => {type: 'SET_NAVIGATION_BLOCKED', blocked: boolean};
         getChannel: (channelId: string) => ActionFunc;
-        getTeam: (teamId: string) => Promise<ActionResult>;
-        getChannelModerations: (channelId: string) => Promise<Array<ChannelPermissions>>;
+        getTeam: (teamId: string) => any;
```

Comment:
Can we change the return type here to `void`? We try to avoid `any` wherever we can.

Reply:
I cannot change`getTeam` to `void`. It would be something like `Promise<ActionResult>`. Problem is that it is incompatible with redux declaration which is about `ActionFunc`... DIdn't find any solution (if I put `ActionFunc|Promise<ActionResult>` I have an error) that's why I put `any` here.

Reply:
Have you tried using `ActionCreatorsMapObject`? It allows you to manually specify the method signatures for all of your bound actions, and it does solve some TS headaches.

We have a few examples in our codebase if you'd like to see how it works.

Reply:
Thx, it helped a lot! I was able to remove all `any` and some weird union types declarations. I have one last (hopefully!) issue with `setNavigationBlocked`:
```
Type '{ type: "SET_NAVIGATION_BLOCKED"; blocked: any; }' provides no match for the signature '(dispatch: DispatchFunc, getState: GetStateFunc): { data: any; } | { error: any; } | Promise<{ data: any; } | { error: any; } | ActionResult[]>'.
```
I'm trying to solve it.

Reply:
Did some kind of hack in aa3997e6b6, but not really convinced. What do you think?